.github/workflows/protocol_test_ci.yml

```diff
@@ -108,22 +29,36 @@ jobs:
 
       - uses: actions/checkout@v2
 
-      - name: Download generated test sdk
-        uses: actions/download-artifact@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Set up JDK
+        uses: actions/setup-java@v1
         with:
-          name: ${{ matrix.protocol }}
-          path: ${{ matrix.protocol }}
+          java-version: ${{ env.java_version }}
 
-      - name: Copy Gemfile
+      - name: Build and publish maven local smithy-ruby
         run: |
-          cp Gemfile ${{ matrix.protocol }}
+          git clone --depth 1 https://github.com/smithy-lang/smithy-ruby.git
+          cd smithy-ruby/codegen
+          ./gradlew clean build publishToMavenLocal
+          cd ../..
```

Comment:
Instead of cd back and forth, can we just set a working dir here?

Reply:
no - the working directory doesn't exist before this